plugins/ImageGraph/API.php

```diff
@@ -129,404 +130,453 @@ public function get(
-            $graph->setFont($font);
-            $graph->setFontSize($fontSize);
-            $graph->setBackgroundColor($backgroundColor);
-            $graph->setTextColor($textColor);
-            $graph->setException($e);
-            $graph->renderGraph();
-        }
+            // restoring get parameters
+            $_GET = $savedGET;
 
-        // restoring get parameters
-        $_GET = $savedGET;
+            switch ($outputType) {
+                case self::GRAPH_OUTPUT_FILE:
+                    if ($idGoal != '') {
+                        $idGoal = '_' . $idGoal;
+                    }
+                    $fileName = self::$DEFAULT_PARAMETERS[$graphType][self::FILENAME_KEY] . '_' . $apiModule . '_' . $apiAction . $idGoal . ' ' . str_replace(',', '-', $date) . ' ' . $idSite . '.png';
+                    $fileName = str_replace(array(' ', '/'), '_', $fileName);
 
-        switch ($outputType) {
-            case self::GRAPH_OUTPUT_FILE:
-                if ($idGoal != '') {
-                    $idGoal = '_' . $idGoal;
-                }
-                $fileName = self::$DEFAULT_PARAMETERS[$graphType][self::FILENAME_KEY] . '_' . $apiModule . '_' . $apiAction . $idGoal . ' ' . str_replace(',', '-', $date) . ' ' . $idSite . '.png';
-                $fileName = str_replace(array(' ', '/'), '_', $fileName);
+                    if (!Filesystem::isValidFilename($fileName)) {
+                        throw new Exception('Error: Image graph filename ' . $fileName . ' is not valid.');
+                    }
 
-                if (!Filesystem::isValidFilename($fileName)) {
-                    throw new Exception('Error: Image graph filename ' . $fileName . ' is not valid.');
-                }
+                    return $graph->sendToDisk($fileName);
 
-                return $graph->sendToDisk($fileName);
+                case self::GRAPH_OUTPUT_PHP:
+                    return $graph->getRenderedImage();
 
-            case self::GRAPH_OUTPUT_PHP:
-                return $graph->getRenderedImage();
+                case self::GRAPH_OUTPUT_INLINE:
+                default:
+                    $pngImgArr[$w]['source'] = $graph->getPngImageData();
+                    $pngImgArr[$w]['id'] = $idSite;
+            }
+        }
+        // render one or more image graphs as HTML page
+        $imageGraphCount = count($pngImgArr);
+        if ($imageGraphCount >= 1) {
+            $this->renderImageGraph($pngImgArr, $imageGraphCount);
+        }
+    }
 
-            case self::GRAPH_OUTPUT_INLINE:
-            default:
-                $graph->sendToBrowser();
-                exit;
+    /**
+     * @param array $pngImgArr Each array index contains image tag with base_64 encoded value of image source
+     * @param string $imageGraphCount Number of img tags available for output
+     * Exits after outputting HTML content
+     */
+    public function renderImageGraph($pngImgArr, $imageGraphCount)
```

Comment:
This should be a private method. Otherwise it will be added to the API documentation as an API